### PR TITLE
refactor: simplify services

### PR DESCRIPTION
**Pull request by vinteo**

### CODE REVIEW

**Comment by travisghansen** [comment]
Still haven't delved into the code but wanted to mention this one `controller.stop_all_stations` as a candidate.

**Comment by Nick-Adams-AU** [comment]
`controller.stop_all_stations` is a great idea!

**Comment by vinteo** [comment]
https://github.com/vinteo/hass-opensprinkler/issues/84

**Comment by vinteo** [comment]
this is ready to go if ok

**Comment by travisghansen** [comment]
@vinteo I think the default values need to be set to `None` and do the whole `if run_seconds is None` business.

The risk we run as is, if something invokes the method with a a non-default value, anything subsequently invoking the method without a default will inherit effectively whatever last changed the value. It's craziness I know.

**Comment by vinteo** [comment]
I removed the defaults instead, and it should be handled by the library now?

**Comment by travisghansen** [comment]
@vinteo I believe you are correct actually. Which is great. LGTM

**Comment by travisghansen** [comment]
@vinteo actually, one last bit, does it make any sense to unify the methods under `OpenSprinklerEntity`? Implement it once everywhere?

**Comment by vinteo** [comment]
I used class inheritance, I think i got it right...

File: custom_components/opensprinkler/__init__.py [hunk 1]
```diff
@@ -293,3 +293,19 @@ class OpenSprinklerSensor(OpenSprinklerEntity):
...
+
+
+class OpenSprinklerProgramEntity:
+    def run(self):
+        """Runs the program."""
+        return self._program.run()
+
+
+class OpenSprinklerStationEntity:
+    def run(self, run_seconds=None):
```
Comment (vinteo) [inline custom_components/opensprinkler/__init__.py:305]: needed to add the `None` for optional

Comment (travisghansen) [reply custom_components/opensprinkler/__init__.py:305]: Looks fantastic.

**Pull request merged**